lib/global-styles.php

```diff
@@ -586,6 +586,24 @@ function gutenberg_experimental_global_styles_flatten_styles_tree( $styles ) {
  * @return string Stylesheet.
  */
 function gutenberg_experimental_global_styles_get_stylesheet( $tree ) {
+
+	// Check if we can use cached.
+	$can_use_cached = (
+		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
+		( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) &&
+		( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) &&
+		! is_admin()
+	);
+
+	if ( $can_use_cached ) {
+
+		// Check if we have the styles already cached.
+		$cached = get_transient( 'global_styles' );
```

Comment:
When we have extensibility we need to document very well what plugins can do. If a plugin changes global styles depending on the page, for example, the caching solution makes things not work well because we have a global cache, so the value will be equal on all pages.